pkg/backends/influxdb/influxdb.go

```diff
@@ -0,0 +1,502 @@
+		counter++
+	})
+	go func() {
+		errs := make([]error, 0, counter)
+	loop:
+		for c := 0; c < counter; c++ {
+			select {
+			case <-ctx.Done():
+				errs = append(errs, ctx.Err())
+				break loop
+			case err := <-results:
+				errs = append(errs, err)
+			}
+		}
+		cb(errs)
+	}()
+}
+
+func (idb *Client) Run(ctx context.Context) {
+	statser := stats.FromContext(ctx).WithTags(gostatsd.Tags{"backend:influxdb"})
+
+	flushed, unregister := statser.RegisterFlush()
+	defer unregister()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-flushed:
+			statser.Gauge("backend.created", float64(atomic.LoadUint64(&idb.batchesCreated)), nil)
+			statser.Gauge("backend.create.failed", float64(atomic.LoadUint64(&idb.batchesCreateFailed)), nil)
+			statser.Gauge("backend.retried", float64(atomic.LoadUint64(&idb.batchesRetried)), nil)
+			statser.Gauge("backend.dropped", float64(atomic.LoadUint64(&idb.batchesDropped)), nil)
+			statser.Gauge("backend.sent", float64(atomic.LoadUint64(&idb.batchesSent)), nil)
+			statser.Gauge("backend.series.sent", float64(atomic.LoadUint64(&idb.seriesSent)), nil)
+		}
+	}
+}
+
+func (idb *Client) processMetrics(ctx context.Context, nowSeconds int64, metrics *gostatsd.MetricMap, cb func(buf *bytes.Buffer, seriesCount uint64)) {
+	fl := flush{
+		timestampSeconds: nowSeconds,
+		flushIntervalSec: idb.flushInterval.Seconds(),
+		metricsPerBatch:  idb.metricsPerBatch,
+		disabledSubtypes: idb.disabledSubtypes,
+		errorCounter:     &idb.batchesCreateFailed,
+		cb:               cb,
+		getBuffer: func() (*bytes.Buffer, io.WriteCloser) {
+			return idb.getBuffer(ctx)
+		},
+		releaseBuffer: idb.releaseBuffer,
+	}
+
+	fl.buffer, fl.writer = fl.getBuffer()
+
+	metrics.Counters.Each(func(metricName, tagsKey string, counter gostatsd.Counter) {
+		if fl.buffer == nil {
+			return
+		}
+		fl.addCounter(metricName, counter.Tags, counter.Value, counter.PerSecond)
+	})
+
+	metrics.Timers.Each(func(metricName, tagsKey string, timer gostatsd.Timer) {
+		if fl.buffer == nil {
+			return
+		}
+		if timer.Histogram == nil {
+			fl.addBaseTimer(metricName, timer)
+		} else {
+			fl.addHistogramTimer(metricName, timer)
+		}
+	})
+
+	metrics.Gauges.Each(func(metricName, tagsKey string, g gostatsd.Gauge) {
+		if fl.buffer == nil {
+			return
+		}
+		fl.addGauge(metricName, g.Tags, g.Value)
+	})
+
+	metrics.Sets.Each(func(metricName, tagsKey string, set gostatsd.Set) {
+		if fl.buffer == nil {
+			return
+		}
+		fl.addSet(metricName, set.Tags, uint64(len(set.Values)))
+	})
+
+	if fl.metricCount == 0 {
+		idb.releaseBuffer(fl.buffer)
+		return
+	}
+
+	fl.finish()
+}
+
+func (idb *Client) postData(ctx context.Context, buffer *bytes.Buffer, seriesCount uint64) error {
+	if err := idb.post(ctx, buffer); err != nil {
+		return err
+	}
+	atomic.AddUint64(&idb.seriesSent, seriesCount)
+	return nil
+}
+
+func writeEvent(w io.Writer, e *gostatsd.Event) {
+	hasHost := false
+	var eventTags []string
+	tags := make(gostatsd.Tags, 0, len(e.Tags)+4)
```

Comment:
Why `+4` on the end here?

Reply:
4 extra tags we expect to add - priority, alerttype, etc.  I can add a comment.